handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java

```diff
@@ -509,32 +500,21 @@ static long toBIO(PrivateKey key) throws Exception {
         }
         ByteBuf buffer = Unpooled.directBuffer();
         try {
-            buffer.writeBytes(BEGIN_PRIVATE_KEY);
-            ByteBuf wrappedBuf = Unpooled.wrappedBuffer(key.getEncoded());
-            final ByteBuf encodedBuf;
-            try {
-                encodedBuf = Base64.encode(wrappedBuf, true);
-                try {
-                    buffer.writeBytes(encodedBuf);
-                } finally {
-                    zerooutAndRelease(encodedBuf);
-                }
-            } finally {
-                zerooutAndRelease(wrappedBuf);
+            PemPrivateKey pem = PemPrivateKey.toPEM(key);
+            buffer.writeBytes(pem.content().slice());
```

Comment:
`PemX509Certificate` and `PemPrivateKey` currently use `Unpooled.buffer()` to allocate the buffer they use for internal storage. However we could change this to `ByteBufAllocator.directBuffer(...)` to make it more likely that `pem.content().isDirect()` is true and thus avoid the copy operation.